tests/fixtures_new.py

```diff
@@ -0,0 +1,240 @@
+import pytest
+from tests import helpers
+from ocs_ci.ocs import constants, ocp
+
+
+@pytest.fixture()
+def secret(request, interface_type):
+    """
+    Create a secret
+
+    Returns:
+        OCS: An instance of secret OCS
+
+    """
+    secrets = list()
+
+    def finalizer():
+        """
+        Delete the RBD secret
+        """
+        for secret in secrets:
+            secret.delete()
+            secret.ocp.wait_for_delete(secret.name)
+
+    request.addfinalizer(finalizer)
+    secrets.append(helpers.create_secret(interface_type=interface_type))
+
+    return secrets
+
+
+@pytest.fixture()
+def pool(request, interface):
+    """
+    Create a pool
+
+    Returns:
+
+    """
+    pools = list()
+
+    def finalizer():
+        """
+        Delete the Ceph block pool
+        """
+        if interface == constants.CEPHBLOCKPOOL:
+            for pool in pools:
+                pool.delete()
+                pool.ocp.wait_for_delete(pool.name)
+
+    request.addfinalizer(finalizer)
+
+    if interface == constants.CEPHBLOCKPOOL:
+        pools.append(helpers.create_ceph_block_pool())
+    elif interface == constants.CEPHFILESYSTEM:
+        pools.append(helpers.get_cephfs_data_pool_name())
+    return pools
+
+
+@pytest.fixture()
+def storageclass(request, secret, pool, interface_type):
+    """
+    Create storage class
+
+    Returns:
+        OCS: An instance of storageclass OCS
+
+    """
+    storageclasses = list()
+
+    def finalizer():
+        """
+        Delete the storage class
+
+        """
+        for sc in storageclasses:
+            sc.delete()
+            sc.ocp.wait_for_delete(sc.name)
+
+    request.addfinalizer(finalizer)
+    storageclasses.append(
+        helpers.create_storage_class(
+            interface_type=interface_type,
+            interface_name=pool[0].name,
```

Comment:
The pool is by default function level scope, so it means that if you will use it in more tests in test class it will in your approach create pool for each storageclass but only first one will be used right?  But as both are function level scope you shouldn't have any other pool than [0] if I understand correctly.

Anyway, what if you would like to create 5 storageclass? I don't think it will work here for one test.